hbase-server/src/main/java/org/apache/hadoop/hbase/io/FSDataInputStreamWrapper.java

```diff
@@ -89,11 +89,6 @@
   // reads without hbase checksum verification.
   private AtomicInteger hbaseChecksumOffCount = new AtomicInteger(-1);
 
-  private Boolean instanceOfCanUnbuffer = null;
```

Comment:
I think these two fields are for performance, where we do not need to do instance of and cast everytime, just test a simple flag.

Reply:
@Apache9 I added the performance flag back. Please review it again! Thanks!